test/test_mcmc_interface.py

```diff
@@ -260,9 +263,56 @@ def model(labels):
 
     kernel = NUTS(model=model)
     mcmc = MCMC(kernel, num_warmup, num_samples, num_chains=num_chains)
+    mcmc.chain_method = chain_method
     init_params = None if not use_init_params else \
         {'coefs': np.tile(np.ones(dim), num_chains).reshape(num_chains, dim)}
     mcmc.run(random.PRNGKey(2), labels, init_params=init_params)
     samples = mcmc.get_samples()
     assert samples['coefs'].shape[0] == 2 * num_samples
     assert_allclose(np.mean(samples['coefs'], 0), true_coefs, atol=0.21)
+
+
+@pytest.mark.parametrize('kernel_cls', [HMC, NUTS])
+@pytest.mark.parametrize('chain_method', [
+    pytest.param('parallel', marks=pytest.mark.xfail(
+        reason='jit+pmap does not work in CPU yet')),
+    'sequential',
+    'vectorized',
+])
+@pytest.mark.skipif('CI' in os.environ, reason="Compiling time the whole sampling process is slow.")
+def test_chain_inside_jit(kernel_cls, chain_method):
+    # NB: this feature is useful for consensus MC.
+    # Caution: compiling time will be slow (~ 90s)
+    if chain_method == 'parallel' and xla_bridge.device_count() == 1:
+        pytest.skip('parallel method requires device_count greater than 1.')
+    warmup_steps, num_samples = 100, 2000
+    # Here are settings which is currently supported.
+    rng = random.PRNGKey(2)
+    step_size = 1.
+    target_accept_prob = 0.8
+    trajectory_length = 1.
+    # Not supported yet:
+    #   + adapt_step_size
```

Comment:
Why are these arguments not supported?

Reply:
For the first two ones, we can support easily by changing if else to lax.cond. It is also doable for max tree depth. However, I dont think that we need to support these features. Jit mcmc is valuable if we want to replicate mcmc with many different settings. To me, the most valuable ones are: step_size, trajectory length as done in many tfp papers (when users want to tune these hyperparameters), or data as in consensus Hmc. For num_warmup, num_samples, supporting them might be complicated because the sample shapes depend on num_samples. I also don’t know if users need to tune these values, so depending on user request, we will consider supporting them.